### ISSUE TITLE
Way to have it work without opening tabs

### ISSUE THREAD

**Issue by 12345Josparky** [DavidKMagnus/projector-hotkeys#5]
I need this but I need it to not open a new tab every time on the PC running OBS. I use it to just assign a hotkey to each scene so it shows that scene on the output monitors for our church but it opens a new tab everytime we switch scenes. Is there a way to have it not open the tabs but set the full projector scene only. The PC running OBS has windows 10 on it.

**Comment by DavidKMagnus** [comment]
This is how we use it as well. Unfortunately there is no way around this. It's a limitation of the OBS API. There is only a function to open a new projector, but not switch to an existing one or close an existing one. I usually end up just closing the extra projectors between services.

One day I would like to try my hand at adding that, but have not had any time to work on it.

**Comment by 12345Josparky** [comment]
Thanks for your hard work and fast reply. To me it seems it should be possible but I know nothing about coding this at all. The reason I say it should be possible is because you can right click the scene and project it to the other screens without it opening the tab. Just seems if this went to the scene then done whatever the right click and choosing the screen does. Maybe I don't understand but seems if you can change it to those screens in OBS without opening the tabs it should be possible to do. Hope that made sense

**Comment by DavidKMagnus** [comment]
The actual OBS interface can do some things that scripts and plugins cannot. Only certain functions are exposed in the API so that scripts and plugins can do them as well, unfortunately!

When you say "tab" do you mean opening another full screen projector window?

**Comment by 12345Josparky** [comment]
Thanks again for your replys. Maybe I am not understanding what this plugin is supposed to do then. What I want is for it to just set Fullscreen projector (source) when the button is pushed for that scene. I don't want it to open the window as well is that possible? It does do that we tested it but we don't want it to open that extra window each time just to set that screen to show which it does do but we get an extra window showing up in the tabs at the bottom that has to be closed or the videos start lagging after so many get opened.

**Comment by DavidKMagnus** [comment]
Yes, that's the part that is beyond my control.
The exposed function that scripts and plugins use can ONLY open new projector windows. There is no way for scripts and plugins to close projectors, to change what's in a projector, or to switch to a projector.

Definitely annoying.

**Comment by 12345Josparky** [comment]
So the plugin is doing what we are wanting it to do but just can't do it like it is done in the program without opening that new tab then correct. So the tab has to be closed every time we change scenes. Looks like OBS would be able to fix this somehow since it can be set in the program without opening the window. Crazy. Again thanks for at least making this now to decide if it is easier to just set it each time from the program or close the extra windows I guess.

**Comment by DavidKMagnus** [comment]
Are you using a Stream deck on Windows?
One thing that might help you is this plugin: https://apps.elgato.com/plugins/com.barraider.windowsmover

You can close windows based on the name. You can make a multi-action to close the old window and then push the hotkey to open a new window.

I used it until switching over to Bitfocus Companion.

**Comment by 12345Josparky** [comment]
cool I will take a look at this thank you for that so it will close the old window without closing the main window when another hotkey is pushed?

**Comment by 12345Josparky** [comment]
apparently I can't use this we don't use elgato?

**Comment by DavidKMagnus** [comment]
Ahh ok, yeah - that's why I asked if you were using a Stream deck. Sorry about that!

**Comment by 12345Josparky** [comment]
Ok so I been playing around with how to get this to work like I want it too and thought I would post what I came up with here.

Basically it has to open that extra tab to show it to the other monitors  that tab has to stay open for that piticular scene. 

So here is what I am thinking. We make one scene with all the sources in it and then I can setup hot keys like we have now but instead of them switching from scene to scene we just have them turn on and off the sources in one scene we create with all the sources. That way we can have it automatically open and project to the extra monitors for that scene and then leave that one extra tab open and switch with the hotkey buttons to the source we want to show. With setting up hotkeys to also close the other scenes just ran a test and you can do that with multiple sources and project the one scene. It is a little work getting the duplicate hotkeys in but I was not even aware till playing around with the hotkeys settings you could even do that but it does work. Might be an option for anyone else looking for this.

**Comment by DavidKMagnus** [comment]
That's a great idea! I'm going to play with that too!
I've seen this plugin before that automatically hides other sources when you make one visible: https://obsproject.com/forum/resources/source-toggler.860/

**Comment by 12345Josparky** [comment]
I don't think that plugin will work for what we are trying to do I don't want the sources for my audio to shut off. I think the hotkeys will work for us. I wasn't aware that you could add extra hotkeys and then duplicate them in the settings till I started playing with it today and it worked lol. It does throw up one of those yellow warnings I guess that is just to let you know you have that same button assigned to something else already but it did work like I wanted it to regardless of that warning ⚠️ sign
